pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -36,9 +43,16 @@ func New(client *kubernetes.Clientset, nodeID, namespace, name, annotation strin
 // Acquire attempts to annotate the kured daemonset with lock info from instantiated DaemonSetLock using client-go
 func (dsl *DaemonSetLock) Acquire(metadata interface{}, TTL time.Duration) (acquired bool, owner string, err error) {
 	for {
-		ds, err := dsl.client.AppsV1().DaemonSets(dsl.namespace).Get(context.TODO(), dsl.name, metav1.GetOptions{})
+		var ds *v1.DaemonSet
+		var lastError error
+		err := wait.PollImmediate(k8sAPICallRetrySleep, k8sAPICallRetryTimeout, func() (bool, error) {
```

Comment:
@squaremo @bboreham thx for the suggestions here; I kinda prefer re-using `wait.PollImmediate`

What do we think about storing the most recently returned error for return (`lastError`) vs just returning the wrapped error from `wait.PollImmediate`? I'm thinking by storing the most recent error we potentially gain more useful failure information, whereas bubbling up the `err` response from `wait.PollImmediate` will potentially just return the equivalent of a context timed out message.

Reply:
> What do we think about storing the most recently returned error for return (lastError)

The last error isn't inherently more useful than any other (in fact, if a context were plumbed through, it would always just be the context timing out). If you feel it might nonetheless have useful information, I would at least wrap the error with a message indicating that retrying timed out.

`context.TODO()` needs to be replaced with a real context. You have two choices:

 1. construct a context for each request, giving each its own timeout (say, the tick duration)
 2. construct a context with the overall timeout and use that for every request.

Reply:
went with option 2, and still have a preference for wrapping the last error we received from the k8s client call, as you point out the error from wait.PollImmediate (at least the way it's being used here) will always be a timeout